addons/account/models/account_journal_dashboard.py

```diff
@@ -71,6 +71,13 @@ def _get_json_activity_data(self):
     json_activity_data = fields.Text(compute='_get_json_activity_data')
     show_on_dashboard = fields.Boolean(string='Show journal on dashboard', help="Whether this journal should be displayed on the dashboard or not", default=True)
     color = fields.Integer("Color Index", default=0)
+    number_vendor_bill_entries = fields.Integer(compute='_compute_number_vendor_bill_entries')
+
+    @api.depends('type')
+    def _compute_number_vendor_bill_entries(self):
+        for journal in self:
+            journal.number_vendor_bill_entries = self.env['account.move'].search_count(
+                [('journal_id', '=', journal.id)]) if journal.type == 'purchase' else 0
```

Comment:
Actually you don't need to count everything.  A field `has_entries` is enough (we can estimate that if there is a refund or something else, the user knows what's up), you can compute it like this (and it is independent from the journal type too):

```python
self.env['account.move'].search([('journal_id', '=', journal.id)], limit=1)
```
This has the advantage that it will stop at the first entry in the journal; and not look up the whole table to count everything.

Or
```python
self.env.cr.execute("""
    SELECT DISTINCT journal_id FROM account_move WHERE journal_id = ANY(%s)
""", [self.ids])
used_ids = {r[0] for r in self.env.cr.fetchall()}
```
This will look up the whole table at worst one time (with possible index optimization still) for all journals.

Or you could also do this
```python
{
    r['journal_id'][0]: r['journal_id_count'] 
    for r in self.env['account.move'].read_group(
        domain=[('journal_id', 'in', self.ids)], 
        fields=['journal_id'], 
        groupby=['journal_id'],
    )
}
```
Same thing as above, except it return the number of entries instead of only the ids where an entry exists. Could be useful in the future and shouldn't have a bigger impact on the perfs than the last proposition.

This might seem like it is not much, but displaying the dashboard needs to be as fast as possible; and the table `account.move` can be quite huge.

Reply:
Thanks a lot for taking the time to explain all this, I really appreciate it :slightly_smiling_face: 
I think the first one is great, since we don't actually need the ids, just that there is at least one entry.
And the code is very nice and clean as well :ok_hand: